### PR TITLE
Particle filter with changing model

### PR DESCRIPTION
**Pull request by havahol**
Main contribution:
- New class WindForcingEnsemble which holds an array of simulators with attached drifters, on which we can do data assimilation. The perturbed parameter is wind direction.
- Notebook that use a particle filter on the WindForcingEnsemble class.
- Upload function for simulator, so that an existing simulator object can be re-initialized by new data from the CPU

Other smaller changes:
- Refactoring Resampling module to DataAssimilationUtils (import as dautils).
- Moving some functions between the dautils module and Drifter classes
- Tests for Drifters are updated accordingly.

### CODE REVIEW

**Comment by havahol** [comment]
[[Milestone] Put together first basic particle filter with changing multiple ocean models](https://trello.com/c/2C6FRLOG/20-milestone-put-together-first-basic-particle-filter-with-changing-multiple-ocean-models)

File: python/tests/dataAssimilation/DrifterTest.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 sys.path.insert(0, '../')
 from SWESimulators import Common
-from SWESimulators import Resampling
```
Comment (setmar) [inline python/tests/dataAssimilation/DrifterTest.py]: Subtests still import Resampling.

File: python/SWESimulators/Common.py [hunk 2]
```diff
@@ -183,6 +183,16 @@ def download(self, cl_queue):
         hv_cpu = self.hv0.download(cl_queue)
         
         return h_cpu, hu_cpu, hv_cpu
+    
+    """
+    Enables downloading data from the additional buffer of CL device to Python
```
Comment (setmar) [inline python/SWESimulators/Common.py]: Docstring should be after definition-line.

File: python/SWESimulators/Common.py [hunk 3]
```diff
@@ -253,6 +263,16 @@ def download(self, cl_queue):
         
         return h_cpu, hu_cpu, hv_cpu
 
+    """
+    Enables downloading data from the additional buffer of CL device to Python
```
Comment (setmar) [inline python/SWESimulators/Common.py]: Docstring should be after definition-line.

**Comment by havahol** [comment]
The commented issues are fixed

**Pull request merged**